### PR TITLE
Update documentation configurations, dependencies, and standards

### PR DESCRIPTION
**Pull request by samjwu**
closes https://github.com/ROCmSoftwarePlatform/hipfort/pull/112

# already done:
	standardization: remove docBin from URL

# done:
	update rocm-docs-core dependency
	add version number to documentation left navigation bar and page title
	add an "About" section with a license page
	enable htmlzip, pdf, epub formats when publishing on Read the Docs
	set pdf title, author, copyright, and version
	standardization: rename .sphinx/.doxygen to sphinx/doxygen
	update dependabot config to include ci:no-build label

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -44,11 +44,17 @@
         ]
     )
 
-docs_core = ROCmDocs("hipfort Documentation")
+getVersionCmd = r'sed -n -e "s/^.*HIPFORT_VERSION.* \"\([0-9\.]\{1,\}\).*/\1/p" ../CMakeLists.txt'
+version = subprocess.getoutput(getVersionCmd)
```
Comment (cgmb) [inline docs/conf.py:48]: How about something directly in Python? If nothing else, it makes it easier to do error checking.

```python
import re

with open('../CMakeLists.txt', encoding='utf-8') as f:
    match = re.search(r'.*\bHIPFORT_VERSION\s+\"?([0-9.]+)[^0-9.]+', f.read())
    if not match:
        raise ValueError("HIPFORT_VERSION not found!")
    version = match[0]
```

Comment (samjwu) [reply docs/conf.py]: Sounds good to me. Done in [use regex for getting hipfort version number](https://github.com/ROCmSoftwarePlatform/hipfort/pull/113/commits/cf4ba174a61bbbafa11ea6ffa81265559cbf794d)

**Review by cgmb — COMMENTED** [review]
I hadn't looked at this too closely the first time, but it's really quite nice.

**Comment by samjwu** [comment]
The failing sles15sp1 check was due to the automatic check.
Since this PR has the ci:docs-ci-only tag, that commit status message will not get updated unfortunately as the GH commit status message API doesn't overwrite everything, just those with the same context.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -6,7 +6,13 @@
...
   - package-ecosystem: "pip" # See documentation for possible values
-    directory: "/docs/.sphinx" # Location of package manifests
+    directory: "/docs/sphinx" # Location of package manifests
     open-pull-requests-limit: 10
     schedule:
       interval: "daily"
+    labels:
+      - "documentation"
+      - "dependencies"
+      - "ci:no-build"
```
Comment (cgmb) [inline .github/dependabot.yml:16]: This needs to be updated. Aside from that, is everything ready on this PR?

Comment (samjwu) [reply .github/dependabot.yml]: Thanks for catching that. Fixed in [7ffaab0](https://github.com/ROCmSoftwarePlatform/hipfort/pull/113/commits/7ffaab08de59add9a0d31c08b1a1d66a21c4ec07)

Yes, this should be ready for review.

**Review by cgmb — APPROVED** [review]

**Pull request merged**